packages/sdk/client-services/src/packlets/services/service-host.ts

```diff
@@ -89,15 +92,26 @@ export class ClientServicesHost {
     }
 
     this._resourceLock = lockKey
-      ? new VaultResourceLock({
-          lockKey,
-          onAcquire: () => {
-            if (!this._opening) {
-              void this.open();
-            }
-          },
-          onRelease: () => this.close(),
-        })
+      ? typeof window !== 'undefined'
+        ? new VaultResourceLock({
+            lockKey,
+            onAcquire: () => {
+              if (!this._opening) {
+                void this.open();
+              }
+            },
+            onRelease: () => this.close(),
+          })
+        : new NodeResourceLock({
+            root: path.join(DX_RUNTIME, 'agent', 'profile', lockKey),
```

Comment:
why not combine root and lockKey into a single param?

browsers really dont care about the format of the string

Reply:
I want to have the lock file and database separately because lock file is runtime data and will be cleared a lot and the database should be persistent

Reply:
Also, I do not want to deal with the issues with browser lock that Josiah described